api/v1alpha1/hardwareclassification_types.go

```diff
@@ -27,8 +27,72 @@ type HardwareClassificationSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of HardwareClassification. Edit HardwareClassification_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// ExpectedHardwareConfiguration defines expected hardware configurations for CPU, RAM, Disk, NIC.
+	ExpectedHardwareConfiguration ExpectedHardwareConfiguration `json:"expectedValidationConfiguration"`
+}
+
+// ExpectedHardwareConfiguration details to match with the host
+type ExpectedHardwareConfiguration struct {
+	// +optional
+	CPU *CPU `json:"CPU"`
+	// +optional
+	Disk *Disk `json:"Disk"`
+	// +optional
+	NIC *NIC `json:"NIC"`
+	// +optional
+	RAM *RAM `json:"RAM"`
+}
+
+// CPU contains CPU details extracted from the hardware profile
+type CPU struct {
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	MinimumCount int `json:"minimumCount" description:"minimum cpu count, greater than 0"`
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	MaximumCount int `json:"maximumCount" description:"maximum cpu count, greater than 0"`
+	// +optional
+	// +kubebuilder:validation:Pattern=`^(0\.\d*[1-9]\d*|[1-9]\d*(\.\d+)?)$`
+	MinimumSpeed string `json:"minimumSpeed" description:"minimum speed of cpu, greater than 0"`
```

Comment:
For floats, you could use resource.Quantity as suggested in https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#type

Alternatively, can the values be expressed as whole numbers using different units?

It isn't clear what units are expected for the field now, so the name should be updated to include that, too.

Reply:
We tried with resource.Quantity but it adds complexity when we are implementing comparison logic.
I am thinking to put it in interger than float like 2600, 2700 (Limit = GHz), because it will clear to everyone if we specify like this.

Reply:
OK, changing the name to something like MinimumKHz (I'm not sure on the capitalization there) would let us use an int.

Do we care about the size of the int? Should we be using int32 or int64 to be more specific? Do we even need something bigger than int16, and would we gain anything from using a smaller size like that?

Reply:
2600 is MHz size, if consider to convirt it into KHz it will be  2600000. I think MHz will be easier to mention in profile as compare to KHz. For MHz, int32 should suffice. It will be more easy for any user.